src/library/walletCollections.js

```diff
@@ -0,0 +1,123 @@
+            .then(wallet => getOpenseaCollection(wallet, collectionName))
+            .then(collection => getAsArray(collection?.nfts));
+    }
+
+    /**
+     * Retrieves metadata for NFTs in a specific collection.
+     * 
+     * @param {string} collectionName - The name of the NFT collection.
+     * @param {string} chainName - The blockchain name (`"ethereum"` or `"polygon"`).
+     * @param {string|null} testWallet - The wallet address to use, or `null` to use the active wallet.
+     * @returns {Promise<Array<Object>>} A promise resolving to an array of metadata objects.
+     */
+    getMetaFromCollection(collectionName, chainName, testWallet) {
+        return this.getNftsFromCollection(collectionName, chainName, testWallet)
+            .then(ownedNfts => {
+                const getNftsMeta = nfts => {
+                    const nftsMeta = [];
+                    const promises = nfts.map(nft =>
+                        new Promise(resolve => {
+                            fetch(nft.metadata_url)
+                                .then(response => response.json())
+                                .then(metadata => {
+                                    nftsMeta.push(metadata);
+                                    resolve();
+                                })
+                                .catch(err => {
+                                    console.warn("Error processing metadata:", nft.metadata_url);
+                                    console.error(err);
+                                    resolve(); // Resolve even on failure to avoid halting
+                                });
+                        })
+                    );
+
+                    return Promise.all(promises).then(() => nftsMeta);
+                };
+
+                return getNftsMeta(ownedNfts);
+            });
+    }
+
+    /**
+     * Retrieves traits or IDs from NFTs in a specific collection.
+     * 
+     * @param {string} collectionName - The name of the NFT collection.
+     * @param {string} chainName - The blockchain name (`"ethereum"` or `"polygon"`).
+     * @param {string} dataSource - The source of the data (`"attributes"` or `"image"`).
+     * @param {string|null} testWallet - The wallet address to use, or `null` to use the active wallet.
+     * @returns {Promise<OwnedTraitIDs>} A promise resolving to an OwnedTraitIDs object.
+     */
+    getTraitsFromCollection(collectionName, chainName, dataSource, testWallet) {
+        return this.getMetaFromCollection(collectionName, chainName, testWallet)
+            .then(nftMeta => new OwnedTraitIDs(nftMeta, dataSource));
+    }
+
+    /**
+     * Switches the active wallet to a specific blockchain and retrieves the wallet address.
+     * 
+     * @param {string} chainName - The blockchain name (`"ethereum"` or `"polygon"`).
+     * @returns {Promise<string>} A promise resolving to the active wallet address, or an empty string on error.
+     */
+    switchWallet(chainName) {
```

Comment:
I feel like this function should be in mint-utils with the other window.ethereum related functions;

You can get the wallet address regardless of the chain; but we need to be on the right chain for minting; so it's relevant there!